test/integration/test_path.py

```diff
@@ -101,3 +103,108 @@ def test_write_delete(backend_aware_bucketfs_params, children_poem, classic_poem
     poem_path1.rm()
     expected_names = {"classic", "highlands.txt"}
     assert _collect_all_names(poems_root) == expected_names
+
+
+def test_infer_path_onprem(backend, backend_aware_bucketfs_params):
+    """
+    Creates the PathLike and validates it.
+    """
+    if backend == "saas":
+        pytest.skip()
+    host_port = re.search(
+        r"http://(\d{1,3}(?:\.\d{1,3}){3}):(\d+)", backend_aware_bucketfs_params["url"]
+    )
+    url = infer_path(
+        bucketfs_host=host_port.group(1),
+        bucketfs_port=host_port.group(2),
```

Comment:
`host_port.group(2),` returns type `str` while function `infer_path()` expects an `int`.
Apparently, mypy / pylint is disabled for tests, which can be OK.



```suggestion
        bucketfs_port=int(host_port.group(2)),
```